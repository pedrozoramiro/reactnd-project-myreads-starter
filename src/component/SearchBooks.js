import React, { Component } from "react";

import Book from "./Book";
import { Link } from "react-router-dom";

class SearchBooks extends Component {
  state = {
    query: ""
  };

  handleChangeSearch = e => {
    const { onSearchBook } = this.props;
    const query = e.target.value;
    this.setState({ query });
    onSearchBook(query);
  };

  render() {
    const { onUpdateShelve ,shelves, books} = this.props;
    const { query } = this.state;

    return (
      <div className="search-books">
        <div className="search-books-bar">
          <Link className="close-search" to="/">
            Close
          </Link>
          <div className="search-books-input-wrapper">
            <input
              type="text"
              placeholder="Search by title or author"
              value={query}
              onChange={this.handleChangeSearch}
            />
          </div>
        </div>
        <div className="search-books-results">
          <ol className="books-grid">
            {books.map(function(book) {
              return (
                <li key={book.id}>
                  <Book
                    book={book}
                    shelves={shelves}
                    onChangeShelve={onUpdateShelve}
                  />
                </li>
              );
            })}
          </ol>
        </div>
      </div>
    );
  }
}
export default SearchBooks;
