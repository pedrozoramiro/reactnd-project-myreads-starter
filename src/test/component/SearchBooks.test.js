import React from "react";
import { shallow, mount } from "enzyme";

import SearchBooks from "../../component/SearchBooks";

import renderer from "react-test-renderer";

import { StaticRouter } from "react-router";

const mock = {
  books: [
    { 
      id:'1',
      title: "Book [read]",
      shelf: "none",
      imageLinks: { smallThumbnail: "" },
      authors: ["Ramiro Pedrozo"]
    },
    {
      id:'2',
      title: "Book [wantToRead]",
      shelf: "wantToRead",
      imageLinks: { smallThumbnail: "" },
      authors: ["Ramiro Pedrozo"]
    }
  ],
  shelves: [
    { name: "currentlyReading", title: "Currently Reading" },
    { name: "wantToRead", title: "Want to Read" },
    { name: "read", title: "Read" }
  ],
  handleChangeShelf: jest.fn(),
  onSearchBook: jest.fn()
};

it("renders without crashing", () => {
  const searchBooksWrapper = (
    <SearchBooks
      books={mock.books}
      shelves={mock.shelves}
      handleChangeShelf={mock.handleChangeShelf}
      onSearchBook={mock.onSearchBook}
    />
  );
  expect(shallow(searchBooksWrapper)).toMatchSnapshot();
});

it("Should call the onSearchBook", () => {
  var context = {};
  const searchBooksWrapper = (
    <StaticRouter location="someLocation" context={context}>
      <SearchBooks
        books={mock.books}
        shelves={mock.shelves}
        handleChangeShelf={mock.handleChangeShelf}
        onSearchBook={mock.onSearchBook}
      />
    </StaticRouter>
  );

  mount(searchBooksWrapper).find("input").simulate('change', { target: { value: 'Hello' } });
  expect(mock.onSearchBook).toHaveBeenCalledTimes(1);
});
