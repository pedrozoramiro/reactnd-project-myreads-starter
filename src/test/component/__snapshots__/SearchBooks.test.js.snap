// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="search-books"
>
  <div
    className="search-books-bar"
  >
    <Link
      className="close-search"
      replace={false}
      to="/"
    >
      Close
    </Link>
    <div
      className="search-books-input-wrapper"
    >
      <input
        onChange={[Function]}
        placeholder="Search by title or author"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="search-books-results"
  >
    <ol
      className="books-grid"
    >
      <li
        key="1"
      >
        <Book
          book={
            Object {
              "authors": Array [
                "Ramiro Pedrozo",
              ],
              "id": "1",
              "imageLinks": Object {
                "smallThumbnail": "",
              },
              "shelf": "none",
              "title": "Book [read]",
            }
          }
          handleChangeShelf={[Function]}
          shelves={
            Array [
              Object {
                "name": "currentlyReading",
                "title": "Currently Reading",
              },
              Object {
                "name": "wantToRead",
                "title": "Want to Read",
              },
              Object {
                "name": "read",
                "title": "Read",
              },
            ]
          }
        />
      </li>
    </ol>
  </div>
</div>
`;
